### ISSUE TITLE
isinstance checks against generic types not working?

### ISSUE THREAD

**Issue by hjwp** [microsoft/pyright#1474]
**Describe the bug**

Pyright can correctly spot `isinstance()` calls which coerce values to a particular type in some cases, but not all -- specifically I found it doesn't work against generic types / `TypeVar`

**To Reproduce**
```python
from typing import List, Union, TypeVar, Type

l = [1, "2", 3]  # type: List[Union[int, str]]

only_ints = [i for i in l if isinstance(i, int)]  # type: List[int]
# ^  this type checks fine

T = TypeVar("T")

def only_some(of_type: Type[T]) -> List[T]:
    return [i for i in l if isinstance(i, of_type)] 
# ^ this does not type check
```

pyright reports this problem:

```
  10:12 - error: Expression of type "list[int | str]" cannot be assigned to return type "List[T@only_some]"
    TypeVar "_T@list" is invariant
      Type "int | str" cannot be assigned to type "T@only_some" (reportGeneralTypeIssues)

```

**Expected behavior**

the `only_some()` function as implemented should type-check ok. fwiw, it does in `mypy`.

**Screenshots or Code**

(see above)

**VS Code extension or command-line**

CLI: 

```
👉  pyright --version
pyright 1.1.109
```

**Additional context**

**Comment by erictraut** [comment]
This is a duplicate of https://github.com/microsoft/pyright/issues/1460. The fix will be in the next release of pyright.

**Comment by hjwp** [comment]
oh i'm so sorry i didn't see that!   and with only like 6 open issues, that's doubly embarassing for me.  heartfelt apologies.

**Comment by erictraut** [comment]
This is addressed in Pyright 1.1.110, which I just published. It will also be included in the next release of Pylance.
